apps/web-giddh/src/app/shared/amount-field/amount-field.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { GeneralService } from '../../services/general.service';
+
+@Component({
+    selector: 'amount-field',
+    templateUrl: './amount-field.component.html',
+    styleUrls: ['./amount-field.component.scss']
+})
+
+export class AmountFieldComponent implements OnInit, OnChanges {
+    @Input() public amount: any;
+    @Input() public currencySymbol: string;
+    @Input() public currencyCode: string;
+    @Input() public ratePrecision: boolean = true;
+
+    /** this will store direction */
+    public direction: string = "ltr";
+
+    constructor(private generalService: GeneralService) {
+
+    }
+
+    public ngOnInit(): void {
```

Comment:
Do add JS doc on all the methods with Document this

Reply:
Done